File: coding-ig/cek-prima/prima.R
cekprima<-function(x){
  # bilangan prima harus > 1
  if (x>1){
    cek <- 1;
    #cek faktor-faktornya
    for (i in seq(2:x)){
      if (i>2 && (x%%i)==0){
        cek<-0;
        cat(x, 'bukan bilangan prima\n');
        cat(i, 'x',floor(x/i),'=',x);
        break;
      }
    }
    if(cek){
      cat(x,'bilangan prima');
    }
  }else{
    # jika x<=1 maka bukan prima
    cat(x, 'bukan bilangan prima');
  }
}
angka<-as.integer(readline(promp="angka = "));
cekprima(angka);

File: coding-ig/cek-prima/prima.c
#include<stdio.h>


int cekprima(int x){
    // bilangan prima harus > 1
    if (x > 1){
        // cek faktor-faktornya
        int cek=1;
        for (int i=2;i<x;i++){
            if ((x%i)==0){
                 cek=0;
                 printf("%d bukan bilangan prima\n",x);
                 printf("%d x %d = %d \n",i,x/i,x);
                 break;
            }
        }if (cek){
            printf("%d bilangan prima",x);
        }     
    } else{
        // jika x <= 1 maka bukan prima
        printf("%d bukan bilangan prima",x);
    
    }
}

int main(){
    int angka;
    printf("angka = ");
    // input dari user
    scanf("%d",&angka);
    // memanggil function cekprima
    cekprima(angka);
}
